### PR TITLE
[Document] Refactor and move source oneCCL to ray start section

### PR DESCRIPTION
**Pull request by Deegue**
closes https://github.com/intel/llm-on-ray/issues/28

### CODE REVIEW

**Comment by xwu99** [comment]
> [Issue#28](https://github.com/intel/llm-on-ray/issues/28)

you can say closes #28 in the above comment then github will link the pr to the issue

**Comment by Deegue** [comment]
> > [Issue#28](https://github.com/intel/llm-on-ray/issues/28)
> 
> you can say "closes #28" in the above comment then github will link the pr to the issue

Updated.

File: docs/setup.md [hunk 1]
```diff
@@ -67,6 +63,10 @@ docker run -it --runtime=habana -v ./llm-on-ray:/root/llm-ray --name="llm-ray-ha
 ```
 
 #### 3. Launch Ray cluster
+If DeepSpeed is enabled, we should dynamic link oneCCL and Intel MPI libraries:
```
Comment (carsonwang) [inline docs/setup.md:66]: This is not only needed when Deepspeed is enabled. When you do distributed finetuning, oneCCL is required.

File: README.md [hunk 2]
```diff
@@ -43,8 +43,6 @@ Software requirement: Python 3.9
 git clone https://github.com/intel/llm-on-ray.git
 cd llm-on-ray
 pip install .[cpu] -f https://developer.intel.com/ipex-whl-stable-cpu -f https://download.pytorch.org/whl/torch_stable.html
-# Dynamic link oneCCL and Intel MPI libraries
```
[the remark this answers is not in the document]
Comment (xwu99) [reply README.md:46]: Yes, just move it to the runtime part
"#### 1. Clone the repository and install dependencies" is the package install part while the source command is for runtime. 
We need to source it for every node before ray start.

**Comment by Deegue** [comment]
Gentle ping @xwu99 for review, thanks!

File: docs/setup.md [hunk 3]
```diff
@@ -67,6 +63,10 @@ docker run -it --runtime=habana -v ./llm-on-ray:/root/llm-ray --name="llm-ray-ha
 ```
 
 #### 3. Launch Ray cluster
+If DeepSpeed is enabled or doing distributed finetuing, oneCCL and Intel MPI libraries should be dynamically linked:
```
Comment (xwu99) [inline docs/setup.md:66]:
```suggestion
If DeepSpeed is enabled or doing distributed finetuing, oneCCL and Intel MPI libraries should be dynamically linked in every node before ray starts:
```

Comment (Deegue) [reply docs/setup.md]: done

**Review by xwu99 — APPROVED** [review]

**Pull request merged**